smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/ReplicationIT.java

```diff
@@ -77,46 +79,50 @@ public static void beforeClass() throws ClientException {
 
     @Before
     public void before() throws TimeoutException, InterruptedException {
-        // Make sure the "publish" distribution agent is present and log its configuration
-        new Polling(() -> {
-            JsonNode agents = adminAuthor.doGetJson(DIST_AGENTS_PATH, 3);
-            log.info("Replication agents list: {}", agents);
-            if (agents.path(PUBLISH_DIST_AGENT).isMissingNode()) {
-                log.warn("Default distribution agent {} is missing from the distribution list", PUBLISH_DIST_AGENT);
-                return false;
-            }
-            return true;
-        }).poll(TIMEOUT, 500);
+        new Polling(this::checkContentDistributionAgentExists)
+        	.poll(TIMEOUT, 500);
     }
 
-    /**
+	/**
      * Activates a page as admin, then deactivates it. Verifies that the page gets removed from publish.
      *
      * @throws Exception if an error occurred
      */
     @Test
     public void testActivateAndDeactivate() throws Exception {
-        rClient.activate(root.getPath());
-        checkPage(SC_OK);
-        rClient.deactivate(root.getPath(), SC_OK);
-        checkPage(SC_NOT_FOUND);
+    	new Polling(this::activateAndDeactivate)
+    		.poll(TIMEOUT, 500);
```

Comment:
@cschneider this basically means retrying the test until the timeout expires. Is this intended? I'm thinking that this could re-trigger the activation several times and I can't tell the side effects of that.

Reply:
I think this could help, but bear in mind it will be the same page uuid. So it's not a clean retry.

Reply:
The idea of the retry is to catch the cases where for some reason during startup the first activate request might be swallowed. 
I did not think about the problem that it is the same page uuid. Do you think this is a problem?